public/js/src/appstore.tag

```diff
@@ -257,43 +259,119 @@ function AppStore(host, services) {
+                    // TODO: Need to determine what to do with the changes array
+                    // ??? RiotControl.trigger('bridge_updated_changes');
+                }
+            );
+        } else {
+            RiotControl.trigger('flash_message', 'error', 'Nothing to save');
+        }
+    });
+
     self.on('shipit_update_value', function (url, value, method) {
-        d('ShipitStore::alter shipment with value:', url, value, method);
+        d('ShipitStore::shipit_update_value', url, value, method);
 
         var payload = {};
 
         if (typeof value === 'object') {
             for (var key in value) {
-                payload[key] = value[key];
+                if (value.hasOwnProperty(key)) {
+                    payload[key] = value[key];
+                }
             }
         } else {
-            console.log('Error: shipit_update_value, Must pass in an object');
+            d('ShipitStore::shipit_update_value(ERROR: must pass in an object, not %s)', typeof value);
             return;
         }
 
-        $.ajax({
-            method: method ? method : 'PUT',
+        // Need to collect the change into an array of changes
+        var change = {
             url: mu(hosts.shipit, 'v1', 'shipment', url),
-            dataType: 'json',
-            contentType: 'application/json; charset=utf-8',
-            accepts: 'application/json',
-            data: JSON.stringify(payload),
+            data: payload,
+            method: method ? method : 'PUT',
             headers: {
                 'x-username': ArgoAuth.getUser(),
                 'x-token': ArgoAuth.getToken()
-            },
-            success: function (result, status, xhr) {
-                d('ShipitStore::shipit_update_value::shipit_update_value_success', url, result);
-                RiotControl.trigger('flash_message', 'success', 'Updated value');
-                RiotControl.trigger('shipit_update_value_result', result);
-            },
-            error: function (xhr, status, err) {
-                var error = xhr.responseText || err || 'Failed To Trigger Shipment!';
-                d('ShipitStore::shipit_update_value_plan::shipit_update_value_error', name, error);
-                RiotControl.trigger('flash_message', 'error', 'Failed to update value ('+ status +')');
-                RiotControl.trigger('shipit_update_value_result', 'error', error);
             }
-        });
+        };
+
+        /*
+         * If there are changes, loop through them
+         * if `change` matches any of them, remove
+         * the one it matches
+         */
+        d('ShipitStore::shipit_update_value(change to be added)', change);
+        d('ShipitStore::shipit_update_value(changes prior to update)', changes);
+        if (changes.length) {
+            changes = changes.filter(function (val, idx) {
+                if (val.url === change.url && val.method === change.method) {
```

Comment:
what if the change is POST and url is /containers or /envVars? This would remove previous calls that would add environment variables. 

For POSTs we need to check if the the name of the data being posted is in the list as well. If not then it will create a new object.